repos/fdbt-site/src/pages/ticketConfirmation.tsx

```diff
@@ -479,12 +479,23 @@ export const buildFlatFareTicketConfirmationElements = (ctx: NextPageContextWith
 
 export const buildSchoolTicketConfirmationElements = (ctx: NextPageContextWithSession): ConfirmationElement[] => {
     const schoolFareTypeAttribute = getSessionAttribute(ctx.req, SCHOOL_FARE_TYPE_ATTRIBUTE) as SchoolFareTypeAttribute;
+    const ticketRepresentationAttribute = getSessionAttribute(
+        ctx.req,
+        TICKET_REPRESENTATION_ATTRIBUTE,
+    ) as TicketRepresentationAttribute;
 
     if (schoolFareTypeAttribute) {
         switch (schoolFareTypeAttribute.schoolFareType) {
             case 'single':
                 return buildSingleTicketConfirmationElements(ctx);
             case 'period':
+                if (!ticketRepresentationAttribute || isWithErrors(ticketRepresentationAttribute)) {
```

Comment:
It doesnt seem right that we're typecasting this on line 485, and then error checking here. Can we not let the code use buildPeriodOrMultiOpTicketConfirmationElements for period tickets (as it currently does) and make our changes inside that function?

Reply:
no changes done in the file